toml/src/main/kotlin/org/rust/toml/CargoTomlPathCompletion.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionProvider
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.completion.DEFAULT_PRIORITY
+import org.rust.lang.core.completion.withPriority
+
+class CargoTomlPathCompletion private constructor(
+    private val target: Target
+) : CompletionProvider<CompletionParameters>() {
+
+    private enum class Target {
+        File, Directory, Any;
+
+        fun priorityOf(file: VirtualFile): Double = when (this) {
+            File -> if (file.isDirectory) DEFAULT_PRIORITY else 1.0
+            Directory -> if (file.isDirectory) 1.0 else DEFAULT_PRIORITY
+            Any -> DEFAULT_PRIORITY
+        }
+    }
+
+    override fun addCompletions(parameters: CompletionParameters, context: ProcessingContext, result: CompletionResultSet) {
+        val keyValue = getClosestKeyValueAncestor(parameters.position) ?: return
+        val containingFile = keyValue.containingFile.originalFile.virtualFile ?: return
+        val path = keyValue.value?.text
```

Comment:
```suggestion
        val path = CompletionUtil.getOriginalElement(keyValue.value)?.text
```

After this change you will be able to remove `?.substringBefore("IntellijIdeaRulezzz")` :)

Reply:
Thank you